crypto/initthread.c

```diff
@@ -421,6 +421,7 @@ static int init_thread_deregister(void *index, int all)
         CRYPTO_THREAD_lock_free(gtr->lock);
         sk_THREAD_EVENT_HANDLER_PTR_free(gtr->skhands);
         OPENSSL_free(gtr);
+        glob_tevent_reg = NULL;
```

Comment:
If we are doing this, I'd prefer if this is moved as early as possible, ideally in the all case it should be done immediately after the get_global_tevent_register(). Also I would expect the all case to not `return 0; `above in case `hands == NULL` and just break out of the loop.
Of course this will not make OPENSSL_cleanup race free, but at least the race window will be as small as possible.